### PR TITLE
Fix broken links.

### PR DESCRIPTION
**Pull request by hzy001**
Signed-off-by: Hao Ziyu <haoziyu@qiyi.com>

### CODE REVIEW

File: demo/README.md [hunk 1]
```diff
@@ -60,9 +60,7 @@ This is a list of short codes introducing different functionalities of xgboost p
 Most of examples in this section are based on CLI or python version.
 However, the parameter settings can be applied to all versions
 
-- [Binary classification](binary_classification)
```
Comment (hcho3) [inline demo/README.md:63]: The page was moved to `CLI/binary_classification`. Can you update this link to `CLI/binary_classification`?

File: demo/README.md [hunk 2]
```diff
@@ -60,9 +60,7 @@ This is a list of short codes introducing different functionalities of xgboost p
...
 - [Multiclass classification](multiclass_classification)
-- [Regression](regression)
```
Comment (hcho3) [inline demo/README.md:65]: The page was moved to `CLI/regression`. Can you update this link to `CLI/regression`?

**Review by hcho3 — CHANGES_REQUESTED** [review]

**Comment by trivialfis** [comment]
Oops, I just had a new branch for this.  Will push on this instead.

**Comment by trivialfis** [comment]
Close https://github.com/dmlc/xgboost/issues/6453 .
Close https://github.com/dmlc/xgboost/issues/6315 .

**Review by hcho3 — APPROVED** [review]

**Pull request merged**